nums/core/array/blockarray.py

```diff
@@ -445,6 +445,11 @@ def tensordot(self, other, axes=2):
         if not isinstance(other, BlockArray):
             raise ValueError("Cannot automatically construct BlockArray for tensor operations.")
 
+        if not isinstance(axes, int):
+            raise TypeError(f"'{type(axes).__name__}' object cannot be interpreted as an integer")
+
+        array_utils.np_tensordot_param_test(self.shape, self.ndim, other.shape, other.ndim, axes)
```

Comment:
Can you rewrite this so that the `array_utils.np_tensordot_param_test` returns a bool, and we raise the error in this function?